### PR TITLE
Add terraform addons schema

### LINKED ISSUE #74: Support schema for engine config and recipes
https://json-schema.org/

### PR DESCRIPTION
**Pull request by umairidris**
Fix #74

### CODE REVIEW

File: templates/tfengine/recipes/terragrunt_deployment.hcl [hunk 1]
```diff
@@ -12,6 +12,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+schema = {
+  title       = "Terraform Deployment Recipe."
```
Comment (MartinPetkov) [inline templates/tfengine/recipes/terragrunt_deployment.hcl:16]: Shouldn't this be Terragrunt?

File: templates/tfengine/recipes/terragrunt_deployment.hcl [hunk 2]
```diff
@@ -12,6 +12,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+schema = {
```
Comment (MartinPetkov) [inline templates/tfengine/recipes/terragrunt_deployment.hcl:15]: This schema looks duplicated. Is there no way to un-duplicate?

Comment (umairidris) [reply templates/tfengine/recipes/terragrunt_deployment.hcl:15]: I think in a follow up PR I will merge the two into one recipe called deployment and that will have a flag for enabling terragrunt.

File: templates/tfengine/recipes/terragrunt_deployment.hcl (hunk 1 above)
Comment (umairidris) [reply templates/tfengine/recipes/terragrunt_deployment.hcl]: Right, fixed

**Pull request merged**